packages/owl-bot/test/database.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it, before} from 'mocha';
+import admin from 'firebase-admin';
+import {FirestoreConfigsStore, encodeId, decodeId} from '../src/database';
+import {Configs} from '../src/configs-store';
+import {v4 as uuidv4} from 'uuid';
+import * as assert from 'assert';
+
+describe('database', () => {
+  before(async function () {
+    // When checks run on github, the call to db.collection().doc().get()
+    // doesn't return an error; it never completes.  So, we need a timeout
+    // to handle that case too.
+    const timeOutPromise = new Promise((resolve, reject) => {
+      setTimeout(() => reject('Timed out!'), 10000);
+    });
+    admin.initializeApp({
+      credential: admin.credential.applicationDefault(),
+    });
+    const db = admin.firestore();
+    try {
+      await Promise.race([
+        timeOutPromise,
+        db.collection('hello').doc('world').get(),
+      ]);
+    } catch (e) {
+      // No connection to firestore.  Not possible to test.
```

Comment:
Perhaps create a `TODO` comment suggesting we mock out some of the `firestore` API, so that we can test in our actions?

Reply:
I think we have more valuable things to do with our limited time.  I wouldn't trust the results anyway.  For example, I never would have discovered the slashes-in-doc-ids issue had I not tested against the real backend.